thrimbletrimmer/scripts/edit.js

```diff
@@ -2156,3 +2196,70 @@ function canEditMetadata() {
+ */
+function createTemplateCropWidgets() {
+	if (videoFrameStage == null) {
+		videoFrameStage = Jcrop.attach('video-info-thumbnail-template-video-source-image');
+		videoFrameStage.listen('crop.update',function(widget,e){
+			const pos = widget.pos;
+			const fieldX1 = document.getElementById("video-info-thumbnail-crop-0");
+			const fieldY1 = document.getElementById("video-info-thumbnail-crop-1");
+			const fieldX2 = document.getElementById("video-info-thumbnail-crop-2");
+			const fieldY2 = document.getElementById("video-info-thumbnail-crop-3");
+			fieldX1.value = pos.x*2;
+			fieldY1.value = pos.y*2;
+			fieldX2.value = (pos.x+pos.w)*2;
+			fieldY2.value = (pos.y+pos.h)*2;
+		});
+	}
+	if (templateStage == null) {
+		templateStage = Jcrop.attach('video-info-thumbnail-template-overlay-image');
+		templateStage.listen('crop.change',function(widget,e){
+			const pos = widget.pos;
+			const fieldX1 = document.getElementById("video-info-thumbnail-location-0");
+			const fieldY1 = document.getElementById("video-info-thumbnail-location-1");
+			const fieldX2 = document.getElementById("video-info-thumbnail-location-2");
+			const fieldY2 = document.getElementById("video-info-thumbnail-location-3");
+			fieldX1.value = pos.x*2;
+			fieldY1.value = pos.y*2;
+			fieldX2.value = (pos.x+pos.w)*2;
+			fieldY2.value = (pos.y+pos.h)*2;
+		});
+	}
+
+	updateTemplateCropWidgets();
+}
+
+/**
+ * Helper function to update the Jcrop widgets (creating them if needed) based
+ * on the current values in the advanced template cropping text input fields.
+ */
+function updateTemplateCropWidgets() {
+	const videoFieldX1 = document.getElementById("video-info-thumbnail-crop-0");
+	const videoFieldY1 = document.getElementById("video-info-thumbnail-crop-1");
+	const videoFieldX2 = document.getElementById("video-info-thumbnail-crop-2");
+	const videoFieldY2 = document.getElementById("video-info-thumbnail-crop-3");
+	const videoFrameRect = Jcrop.Rect.create(videoFieldX1.value/2, videoFieldY1.value/2, (videoFieldX2.value-videoFieldX1.value)/2, (videoFieldY2.value-videoFieldY1.value)/2);
+	if (videoFrameStage.active == null) {
+		videoFrameStage.newWidget(videoFrameRect);
+	} else {
+		videoFrameStage.active.pos = videoFrameRect;
+		videoFrameStage.active.render();
+	}
+
+	const templateFieldX1 = document.getElementById("video-info-thumbnail-location-0");
+	const templateFieldY1 = document.getElementById("video-info-thumbnail-location-1");
+	const templateFieldX2 = document.getElementById("video-info-thumbnail-location-2");
+	const templateFieldY2 = document.getElementById("video-info-thumbnail-location-3");
+	const templateRect = Jcrop.Rect.create(templateFieldX1.value/2, templateFieldY1.value/2, (templateFieldX2.value-templateFieldX1.value)/2, (templateFieldY2.value-templateFieldY1.value)/2);
+	if (templateStage.active == null) {
+		templateStage.newWidget(templateRect);
+	} else {
+		templateStage.active.pos = templateRect;
+		templateStage.active.render();
+	}
+}
```

Comment:
Please ensure your editor adds trailing newlines to files.
```suggestion
}
```